code/io/timer.cpp

```diff
@@ -179,6 +179,39 @@ UI_TIMESTAMP ui_timestamp_delta(UI_TIMESTAMP stamp, int delta_ms)
 	return UI_TIMESTAMP(stamp.value() + delta_ms);
 }
 
+// Returns the difference between two timestamps as an int, does not check that the result will be positive.
+// Also only error check if a timestamp is invalid, but when either is immediate or never, it returns 0.
+int timestamp_get_delta(TIMESTAMP before, TIMESTAMP after)
+{
+	Assertion(before.isValid(), "timestamp_get_delta called with an invalid before timestamp%s. This is a coder mistake, please report!", (after.isValid()) ? "" : " and an invalid after timestamp.");
```

Comment:
Why add the check for the 'after' timestamp here, given that the next line checks it anyway?

Reply:
Because the after timestamp could be invalid as well, and it's helpful to know whether both are invalid instead of just one.

Reply:
Yeah but literally the very next line will let you know.

Reply:
Except that literally the next line will not trigger on the same run if the first one does.

Reply:
Unless you deliberately continue, but yeah, I see your point.